frontend/packages/pipelines-plugin/src/components/pipelines/detail-page-tabs/pipeline-details/PipelineVisualizationTask.tsx

```diff
@@ -261,8 +263,8 @@ const TaskComponent: React.FC<TaskProps> = ({
         <WhenExpressionDecorator
           width={WHEN_EXPRESSSION_DIAMOND_SIZE}
           height={WHEN_EXPRESSSION_DIAMOND_SIZE}
-          stroke={showStatusState ? taskColor : undefined}
-          color={taskColor}
+          stroke={showStatusState ? whenExpressionDiamondColor : undefined}
+          color={whenExpressionDiamondColor}
```

Comment:
This is not added by you, but I don't understand why we make the color calculation outside and the tooltip text decision in `WhenExpressionDecorator`.

I'm not sure, but I think it's possible to remove stroke and color props here and calculate these in the `switch (status)` the component already have. You maybe need a new prop for `isFinallyTask={isFinallyTask}`

@debsmita1 @karthikjeeyar @andrewballantyne Wdyt? Is there an edge case I don't see yet? Or is there any reason to calculate these data on different component levels?